ext/mimemagic/Rakefile

```
@@ -0,0 +1,33 @@
+require "rubygems"
+require "rake/clean"
+
+def locate_mime_database
+  # User provided path.
+  unless ENV["FREEDESKTOP_MIME_TYPES_PATH"].nil?
+    path = File.expand_path(ENV["FREEDESKTOP_MIME_TYPES_PATH"])
+    unless File.exist?(path)
+      raise "The path #{path} was provided for the MIME types database, but no file exists at that path."
+    end
+    return path
+  end
+
+  # Default path on Linux installs for the MIME types database.
+  return "/usr/share/mime/packages/freedesktop.org.xml" if File.exist?("/usr/share/mime/packages/freedesktop.org.xml")
+
+  # Default path when installed with Homebrew
+  return "/usr/local/Cellar/shared-mime-info/2.1/share/shared-mime-info/packages/freedesktop.org.xml" if File.exist?("/usr/local/Cellar/shared-mime-info/2.1/share/shared-mime-info/packages/freedesktop.org.xml")
```

On a Mac this is specifically looking at the directory of the 2.1 version of the Homebrew Formula which could become outdated. In addition `/usr/local` will not necessarily be the correct `prefix` there.

Specifically in Homebrew 3.0, Intel macs use `/usr/local` and ARM macs use `/opt/homebrew`. I just tested on default installs on both architectures, and I am able to get the following non-versioned paths:

Intel: `/usr/local/share/mime/packages/freedesktop.org.xml`
ARM: `/opt/homebrew/share/mime/packages/freedesktop.org.xml`

(Those are symlinks to the versioned path currently in here)

Homebrew lets you use `brew --prefix` to get the actual prefix, but that may be asking too much here. I think a reasonable solution though, might be to at least check for both of these paths.


```suggestion
  return "/usr/local/share/mime/packages/freedesktop.org.xml" if File.exist?("/usr/local/share/mime/packages/freedesktop.org.xml")
  return "/opt/homebrew/share/mime/packages/freedesktop.org.xml" if File.exist?("/opt/homebrew/share/mime/packages/freedesktop.org.xml")
```